[FIX] point_of_sale,base: use context timezone for sequence

nle-odoo opened the pull request:
Previously when replacing time related sequence in a prefix or suffix of
a sequence, the timezone used for the time values would be the server's
timezone.

With this fix, the context timezone is used if present, UTC is used
otherwise.

opw-646487

beledouxdenis commented:
1. Instead of reinventing the wheel, can't you use `context_timestamp` ?
`dt = context_timestamp(cr, uid, datetime.now(), context=context)`
2. Even if `_` methods are not supposed to be overridden, this isn't impossible that someone did, for instance to solve this issue by their own. Changing the signature would therefore break their override.
Therefore, I suggest you rename this 
`def _interpolation_dict(self, context=None):`
by 
`def _interpolation_dict_context(self, context=None):`

and then create under this method a new method:
```
def _interpolation_dict(self):
    return self._interpolation_dict_context(context=None)
```
and then change the call
`d = self._interpolation_dict(context=context)`
to
`d = self._interpolation_dict_context(context=context)`

nle-odoo commented:
@dle i updated the pull request

I could not call ` return self._interpolation_dict_context(context=None)` like this since we are in the old api, so I added a parameter `timestamp`. but I don't know if this is better than duplicating code since:

:+1:  with this when an external code call `_interpolation_dict`, external override of `_interpolation_dict` and `_interpolation_dict_context` will be called

:-1:  but if an external override of `_interpolation_dict_context` expects to have a real `cr` issue will happen.

beledouxdenis commented:
Well, you are right. I did not think about the fact `context_timestamp` needed `cr` and `uid`.

I guess we can then use the `pytz` part of your first fix, but still keep these two `_interpolate_dict...` methods for the retro-compatibility :).

Thank you !

nle-odoo commented:
just to be sure, is it okay like this? 

it seems better to me to use context_timestamp.

beledouxdenis commented:
Well, it could have been better if cr and uid were available in `interpolation_dict`, but this is not the case.

I would rather like what you did first, with the pytz, and not context_timestamp, just because of this `cr, uid` thing.

nle-odoo commented:
ok, thanks, so because we prefer to not disable existing overriding.

I will push when green ^^

beledouxdenis commented:
Seems good :) thank you !
